main.go

```diff
@@ -1006,46 +1019,45 @@ func _main(ctx context.Context) error {
 
 	explore.StartMempoolMonitor(notify.NtfnChans.ExpNewTxChan)
 
-	if cfg.MonitorMempool {
-		// Create the mempool data collector.
-		mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
-		if mpoolCollector == nil {
-			return fmt.Errorf("Failed to create mempool data collector")
-		}
-
-		// Collect and store initial mempool data.
-		mpData, err := mpoolCollector.Collect()
-		if err != nil {
-			return fmt.Errorf("Mempool info collection failed while gathering"+
-				" initial data: %v", err.Error())
-		}
-
-		// Store initial MP data.
-		if err = baseDB.MPC.StoreMPData(mpData, time.Now()); err != nil {
-			return fmt.Errorf("Failed to store initial mempool data (WiredDB): %v",
-				err.Error())
-		}
-
-		// Setup the mempool monitor, which handles notifications of new
-		// transactions.
-		mpi := &mempool.MempoolInfo{
-			CurrentHeight:               mpData.GetHeight(),
-			NumTicketPurchasesInMempool: mpData.GetNumTickets(),
-			NumTicketsSinceStatsReport:  0,
-			LastCollectTime:             time.Now(),
-		}
+	// Create the mempool data collector.
+	mpoolCollector := mempool.NewMempoolDataCollector(dcrdClient, activeChain)
+	if mpoolCollector == nil {
+		return fmt.Errorf("Failed to create mempool data collector")
+	}
 
-		// Parameters for triggering data collection. See config.go.
-		newTicketLimit := int32(cfg.MPTriggerTickets)
-		mini := time.Duration(cfg.MempoolMinInterval) * time.Second
-		maxi := time.Duration(cfg.MempoolMaxInterval) * time.Second
+	// Collect and store initial mempool data.
+	stakeData, txs, err := mpoolCollector.Collect()
+	if err != nil {
+		return fmt.Errorf("Mempool info collection failed while gathering"+
+			" initial data: %v", err.Error())
+	}
 
-		mpm := mempool.NewMempoolMonitor(ctx, mpoolCollector, mempoolSavers,
-			notify.NtfnChans.NewTxChan, &wg, newTicketLimit, mini, maxi, mpi)
-		wg.Add(1)
-		go mpm.TxHandler(dcrdClient)
+	// Store initial mempool data.
+	for _, s := range mempoolSavers {
+		s.StoreMPData(stakeData, txs)
 	}
 
+	// The MempoolMonitor receives notifications of new transactions on
+	// notify.NtfnChans.NewTxChan, and of new blocks on the same channel with a
+	// nil transaction message. The mempool monitor will process the
+	// transactions, and forward new ones on via the mpDataToPSHub with an
+	// appropriate signal to the underlying WebSocketHub on signalToPSHub.
+	signalToPSHub, mpDataToPSHub := psHub.HubRelays()
+	mempoolSigOuts := []chan pstypes.HubSignal{signalToPSHub}
+	newTxOuts := []chan *exptypes.MempoolTx{mpDataToPSHub}
```

Comment:
psHub is the only data receiver, but these are slices of channels to allow multiple receivers of data from MempoolMonitor.